store/cachekv/store.go

```diff
@@ -54,32 +33,24 @@ func (store *Store) GetStoreType() types.StoreType {
 }
 
 // Get implements types.KVStore.
-func (store *Store) Get(key []byte) (value []byte) {
-	store.mtx.Lock()
-	defer store.mtx.Unlock()
-
+func (store *Store) Get(key []byte) []byte {
 	types.AssertValidKey(key)
 
-	cacheValue, ok := store.cache[conv.UnsafeBytesToStr(key)]
-	if !ok {
-		value = store.parent.Get(key)
-		store.setCacheValue(key, value, false, false)
-	} else {
-		value = cacheValue.value
+	if value, found := store.cache.Get(key); found {
+		return value
 	}
 
+	value := store.parent.Get(key)
+	store.setCacheValue(key, value, false)
```

Comment:
What happens if the key is not in the parent store?

Reply:
It'll cache the nil value.

Reply:
Would make sense to avoid caching the nil value by checking the value returned by `store.parent.Get(key)`?

Reply:
I think the non-existence result is catchable as well?

Reply:
Makes sense, thanks!